content/blog/integrating-embedded-swift-and-sdl3-on-esp32/index.md

```diff
@@ -0,0 +1,443 @@
+﻿---
+title: "Integrating Embedded Swift and SDL3 on ESP32"
+date: 2024-11-22
+showAuthor: false
+authors:
+  - "juraj-michalek"
+tags: ["ESP32-P4", "ESP32-C6", "ESP32-C3", "Swift", "CI/CD",  "SDL", "Graphics", "Filesystem", "Experimental"]
+---
+
+The article explains how to build graphical application with Embedded Swift and SDL3 on ESP32-P4, ESP32-C3, ESP32-C6.
+The full project is available at [GitHub repository](https://github.com/georgik/esp32-sdl3-swift-example).
+
+## Introduction
+
+[Embedded Swift](https://www.swift.org/blog/embedded-swift-examples/) brings the power and elegance of the Swift programming language to embedded systems like the ESP32 microcontrollers. In this article, we'll explore how to develop graphical applications for the ESP32-C3 and ESP32-C6 using Swift. We'll demonstrate how to integrate C libraries like SDL3 (Simple DirectMedia Layer 3), commonly used in desktop applications, into your Swift projects. With the help of the ESP-IDF (Espressif IoT Development Framework) and ESP-BSP (Board Support Package), you can create cross-platform applications that run on various ESP32 boards with minimal effort.
```

Comment:
Here, it will be nice to give a short introduction to Swift vs Embedded Swift.
This introduction and the next section are similar, consider joining both or changing the introduction not to mention ESP32 yet.